presto-main/src/main/java/io/prestosql/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -852,21 +852,28 @@ else if (equality.getRight().equals(column) && columns.contains(equality.getLeft
 
```

Comment:
> This would result in remote exchanges being omitted, causing queries
to return incorrect results.

Could we add a test to ATQ or to `TestFullOuterJoinWithCoalesce` where we would also validate query results?